cli/tools/doc.rs

```diff
@@ -25,21 +27,53 @@ pub async fn print_docs(
   let source_file = doc_flags
     .source_file
     .unwrap_or_else(|| "--builtin".to_string());
-  let source_parser = deno_graph::DefaultSourceParser::new();
 
   let mut doc_nodes = if source_file == "--builtin" {
+    struct LibDenoDtsLoader {
```

Comment:
This should be changed to use a `deno_graph::MemoryLoader` actually. I will do that tomorrow along with adding unit tests for the cache.